File: 43 Multiply Strings/MultiplyStrings.java
class MultiplyStringsœ {
    public String multiply(String num1, String num2) {
        int m = num1.length();
        int n = num2.length();
        int[] product = new int[m + n];
        for (int i = m - 1; i >= 0; i--) {
            for (int j = n - 1; j >= 0; j--) {
                int p1 = i + j;
                int p2 = i + j + 1;
                int sum = (num1.charAt(i) - '0') * (num2.charAt(j) - '0') + product[p2];
                product[p1] += sum / 10;
                product[p2] = sum % 10;
            }
        }
        StringBuilder res = new StringBuilder();
        for (int i = 0; i < m + n; i++) {
            if (res.length() == 0 && product[i] == 0) {
                continue;
            } 
            res.append(product[i]);
        }
        return res.length() == 0? "0": res.toString();
    }
}

File: 0043 Multiply Strings/multiply_strings.py
class Solution:
    def multiply(self, num1: str, num2: str) -> str:
        m = len(num1)
        n = len(num2)
        res = [0] * (m + n)
        for i in range(m - 1, -1, -1):
            for j in range(n - 1, -1, -1):
                p1 = i + j
                p2 = i + j + 1 
                pro = (ord(num1[i]) - 48) * (ord(num2[j]) -48) + res[p2]
                res[p2] = pro % 10
                res[p1] += pro // 10
        result = ""
        for i in res:
            if i == 0 and result == "":
                continue
            else:
                result += str(i)
        return "0" if result == "" else result